src/interpolation/details/ArborX_InterpDetailsCompactRadialBasisFunction.hpp

```diff
@@ -0,0 +1,88 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_INTERP_DETAILS_COMPACT_RADIAL_BASIS_FUNCTION_HPP
+#define ARBORX_INTERP_DETAILS_COMPACT_RADIAL_BASIS_FUNCTION_HPP
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX::Interpolation
+{
+
+namespace Details
+{
+
+template <typename T, std::size_t N>
+KOKKOS_INLINE_FUNCTION T evaluatePolynomial(T const x, T const (&coeffs)[N])
+{
+  T eval = 0;
+  for (std::size_t i = 0; i < N; i++)
+    eval = x * eval + coeffs[i];
+  return eval;
+}
+
+} // namespace Details
+
+namespace CRBF
+{
+
+#define CRBF_DECL(NAME)                                                        \
+  template <std::size_t>                                                       \
+  struct NAME;
+
+#define CRBF_DEF(NAME, N, FUNC)                                                \
+  template <>                                                                  \
+  struct NAME<N>                                                               \
+  {                                                                            \
+    template <typename T>                                                      \
+    KOKKOS_INLINE_FUNCTION static constexpr T evaluate(T const y)              \
+    {                                                                          \
+      T const x = Kokkos::min(Kokkos::abs(y), T(1));                           \
+      return Kokkos::abs(FUNC);                                                \
+    }                                                                          \
+  };
+
+#define CRBF_POLY(...) Details::evaluatePolynomial<T>(x, {__VA_ARGS__})
+#define CRBF_POW(X, N) Kokkos::pow(X, N)
+
+CRBF_DECL(Wendland)
```

Comment:
Let's add a link to some place that defines these CRBF, in case we want to add more in the future, and as a place to check.

Reply:
Please address this.

Reply:
As they were taken from DTK I do not know the exact origin of those functions. Those are the supposed sources, but I would need confirmation before adding them:
- Wendland: https://doi.org/10.1016/S1270-9638(00)01087-7 (I do not have access to the paper)
- Wu: https://doi.org/10.1007/BF03177517 (They are defined so this is the source)
- Buhmann: https://doi.org/10.48550/arXiv.1606.09527 (likely correspond to the result of some of the integrals)

Reply:
A better Wendland paper is https://doi.org/10.1007/BF02123482. However, the shape seems to depend on the dimension. In addition, in Wu's paper, there's some parameter $l$. I also didn't see any derivations of Buhmann, so we are essentially are using the magic formulas from DTK-2.0.

So, I think it's worthwhile to understand
- Whether Wendland functions should be adjusted for dimension?
- Whether Wu's functions correspond to some subfamily?
- The derivation of Buhmann

We don't need to do it right now, but I think a comment with a `FIXME` would be appropriate.